File: test90_out/local.mcu/Pin2/Pin2.c

/*
 * ======== string literals ========
 */

static const char local_mcu_Pin2__s__fileName[] = "Pin2.p";
static const char local_mcu_Pin2__s__2[] = ".set\n";
static const char local_mcu_Pin2__s__9[] = ".isOutput\n";
static const char local_mcu_Pin2__s__6[] = ".makeInput\n";
static const char local_mcu_Pin2__s__3[] = ".clear\n";
static const char local_mcu_Pin2__s__5[] = ".get\n";
static const char local_mcu_Pin2__s__0[] = "Pin Init: ";
static const char local_mcu_Pin2__s__7[] = ".isInput\n";
static const char local_mcu_Pin2__s__11[] = "Pin";
static const char local_mcu_Pin2__s__4[] = ".toggle\n";
static const char local_mcu_Pin2__s__8[] = ".makeOutput\n";
static const char local_mcu_Pin2__s__1[] = "\n";
static const char local_mcu_Pin2__s__10[] = "Pin2";

/*
 * ======== function definitions ========
 */

void local_mcu_Pin2_set__E() {
    (local_mcu_Pin2_state__V) = true;
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin2_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin2__s__2);
}

void local_mcu_Pin2_makeOutput__E() {
    (local_mcu_Pin2_input__V) = false;
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin2_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin2__s__8);
}

bool local_mcu_Pin2_isInput__E() {
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin2_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin2__s__7);
    return(  (local_mcu_Pin2_input__V) == true  );
}

bool local_mcu_Pin2_isOutput__E() {
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin2_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin2__s__9);
    return(  (local_mcu_Pin2_input__V) == false  );
}

bool local_mcu_Pin2_get__E() {
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin2_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin2__s__5);
    return(  (local_mcu_Pin2_state__V)  );
}

void local_mcu_Pin2_clear__E() {
    (local_mcu_Pin2_state__V) = false;
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin2_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin2__s__3);
}

void local_mcu_Pin2_toggle__E() {
    (local_mcu_Pin2_state__V) = !(local_mcu_Pin2_state__V);
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin2_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin2__s__4);
}

void local_mcu_Pin2_targetInit__I() {
    (local_mcu_Pin2_name__V) = (local_mcu_Pin2_pname);
    (local_mcu_Pin2_num__V) = (local_mcu_Pin2_pnum);
    (local_mcu_Pin2_input__V) = false;
    (local_mcu_Pin2_state__V) = false;
    test90_Test_03_pollen__printStr((string)local_mcu_Pin2__s__0);
	test90_Test_03_pollen__printStr((string)(local_mcu_Pin2_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin2__s__1);
}

void local_mcu_Pin2_makeInput__E() {
    (local_mcu_Pin2_input__V) = true;
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin2_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin2__s__6);
}

File: test90_out/local.mcu/Pin2/Pin2.js

/*
 * ======== MODULE local.mcu.Pin2 ========
 */

var $$c = function() {
    const Pin2 = {};
    Pin2.Pin2 = Pin2
    Pin2.$name = 'local.mcu.Pin2';
    Pin2.pollen$used = 0;
    Pin2.targetInit = new $$Ref('local_mcu_Pin2_targetInit__I');
    Pin2.set = new $$Ref('local_mcu_Pin2_set__E');
    Pin2.clear = new $$Ref('local_mcu_Pin2_clear__E');
    Pin2.toggle = new $$Ref('local_mcu_Pin2_toggle__E');
    Pin2.get = new $$Ref('local_mcu_Pin2_get__E');
    Pin2.makeInput = new $$Ref('local_mcu_Pin2_makeInput__E');
    Pin2.isInput = new $$Ref('local_mcu_Pin2_isInput__E');
    Pin2.makeOutput = new $$Ref('local_mcu_Pin2_makeOutput__E');
    Pin2.isOutput = new $$Ref('local_mcu_Pin2_isOutput__E');
    Pin2.pname = "Pin";
    Pin2.pnum = 2;
    Pin2.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    Pin2.pollen__uses__ = function() {
    }
    Pin2.$$privateInit = function() {
        Pin2.name = null;
        Pin2.num = 0;
        Pin2.state = 0;
        Pin2.input = 0;
        Pin2.pollen__unitname = "Pin2";
    }
    return Pin2;
}
var $$u = $$c();
$units['local.mcu.Pin2'] = $$u;
$units.push($$u);

